packages/search/src/tokenize.ts

```diff
@@ -59,23 +59,55 @@ export function words(content: string): string[] {
   return content.match(/\b\w+\b/g) ?? [];
 }
 
+/**
+ * Prepares a string for tokenization by splitting it into batches of lines, each of which is
+ * no longer than the specified maximum length.
+ *
+ * @param content The content to split into batches
+ * @param batchSize The maximum number of characters per batch
+ * @param maxLineLength The maximum length of a line
+ * @returns an array of batches of content
+ */
+export function batch(content: string, batchSize = 1000, maxLineLength = 1000): string[] {
+  const lines = content.split('\n').filter(({ length }) => length <= maxLineLength);
+  const result = [];
+  for (let i = 0; i < lines.length; i += batchSize) {
+    result.push(lines.slice(i, i + batchSize).join('\n'));
+  }
+
+  return result;
+}
```

Comment:
Have you considered implementing this as an async generator instead? IMO it'll make the flow clearer and you won't have to muck with the event loop directly.

Reply:
If you're generating batches asynchronously then it goes through the event loop, so you don't need `setImmediate`. Ie.
```typescript
async *batch(): AsyncGenerator<string> {
  for (...) yield chunk;
}

for await (const chunk of batch())
  doSomething(chunk);
```